### ISSUE TITLE
Prevent invalid positions from freezing remote clients

### ISSUE THREAD

**Issue by johnshaughnessy** [mozilla/hubs#3355]
**Description**
If a user sets the position of an object they own to an invalid value, they will send those invalid values over the network and freeze other peoples' clients. 

**To Reproduce**
```
document.querySelector("#avatar-rig").object3D.position.set("foo")
```

**Expected behavior**
It should not be possible to freeze other peoples' clients by sending invalid transform data.

It might be acceptable to freeze your own client. It depends on how costly mitigating this would be.

**Additional context**
Reported by utopiah.

**Possible Solution**
I don't know what the best fix for this is. Perhaps we should replace 
https://github.com/MozillaReality/networked-aframe/blob/19a5a972d5a88423b3e17a12864627d49a6a77d5/src/components/networked.js#L260-L268

With 
```js
        if (componentNames.includes('position')) {
          const position = buffer.getPosition();
          isValidVector3(position) && position.copy();
        }
        if (componentNames.includes('rotation')) {
          const quaternion = buffer.getQuaternion();
          isValidQuaternion(quaternion) && object3D.quaternion.copy(quaternion);
        }
        if (componentNames.includes('scale')) {
          const scale = buffer.getScale();
          isValidVector3(scale) && object3D.scale.copy(scale);
        }
```

Or make sure these `getPosition/Quaternion/Scale` functions always return valid values.

**Comment by Utopiah** [comment]
> Or make sure these getPosition/Quaternion/Scale functions always return valid values.

I don't think it's sufficient since, like in your example, someone can `set()` arbitrary invalid values.

**Comment by johnshaughnessy** [comment]
> someone can set() arbitrary invalid values.

That's true for objects on the local client. I did not suggest a fix for this because I have not thought of one. (I'm wary about inserting something like an `isValid` check in a `setter` or a `getter` on `Object3D` itself for fear of slowing down frequently used codepaths.) That's what I meant by

> It might be acceptable to freeze your own client. It depends on how costly mitigating this would be.

I think the fix I suggested would at least prevent one client from breaking _other peoples'_ clients, which seems like a more serious issue than someone breaking the local client. 

(For clarity -- those `getPosition/Quaternion/Scale` are here: https://github.com/InfiniteLee/buffered-interpolation/blob/3a51af3a7ae519e16d620300b2bb07dc7a3a1bb8/index.js#L180-L190 These could perform the `isValid` instead of having them in the networked-aframe update linked above.)

**Comment by Utopiah** [comment]
Ah yes definitely and indeed that would in most cases be enough. If someone "decides" to break their own client by experimenting, so be it, but at least it shouldn't break others.

**Comment by johnshaughnessy** [comment]
This change to networked-aframe seems to fix the issue for transform updates. Other non-validated properties may break in the same way but transforms are a good place to start. We should make sure this doesn't slow down the client. https://github.com/MozillaReality/networked-aframe/pull/43

**Comment by brianpeiris** [comment]
Fixed with #3613
